src/infra/ssl/sslCmdRepo.go

```diff
@@ -2,135 +2,153 @@ package sslInfra
-	sslQueryRepo := SslQueryRepo{}
+func (repo SslCmdRepo) forceSymlink(
+	pkiSourcePath string,
+	pkiTargetPath string,
+) error {
+	err := os.Remove(pkiTargetPath)
+	if err != nil && !os.IsNotExist(err) {
+		return errors.New("FailedToDeletePkiFile: " + err.Error())
+	}
 
-	vhostConfigFilePath, err := sslQueryRepo.GetVhostConfigFilePath(addSslPair.Hostname)
+	err = os.Symlink(pkiSourcePath, pkiTargetPath)
 	if err != nil {
-		return err
+		return errors.New("AddPkiSymlinkError: " + err.Error())
 	}
 
-	sslBaseDirPath := "/speedia/pki/" + addSslPair.Hostname.String()
-	sslKeyFilePath := sslBaseDirPath + "/ssl.key"
-	sslCertFilePath := sslBaseDirPath + "/ssl.crt"
+	return nil
+}
+
+func (repo SslCmdRepo) ReplaceWithSelfSigned(vhost valueObject.Fqdn) error {
+	vhostStr := vhost.String()
 
-	err = infraHelper.MakeDir(sslBaseDirPath)
-	if err != nil {
-		return err
+	vhostCertFilePath := pkiConfDir + "/" + vhostStr + ".crt"
+	vhostCertFileExists := infraHelper.FileExists(vhostCertFilePath)
+	if vhostCertFileExists {
+		err := os.Remove(vhostCertFilePath)
+		if err != nil {
+			return errors.New("FailedToDeleteCertFile: " + err.Error())
+		}
 	}
 
-	err = infraHelper.UpdateFile(sslCertFilePath, addSslPair.Certificate.String(), true)
-	if err != nil {
-		return err
+	vhostCertKeyFilePath := pkiConfDir + "/" + vhostStr + ".key"
+	vhostCertKeyFileExists := infraHelper.FileExists(vhostCertKeyFilePath)
+	if vhostCertKeyFileExists {
+		err := os.Remove(vhostCertKeyFilePath)
+		if err != nil {
+			return errors.New("FailedToDeleteCertKeyFile: " + err.Error())
+		}
 	}
 
-	err = infraHelper.UpdateFile(sslKeyFilePath, addSslPair.Key.String(), true)
-	if err != nil {
-		return err
+	return infraHelper.CreateSelfSignedSsl(vhostStr)
+}
+
+func (repo SslCmdRepo) Add(addSslPair dto.AddSslPair) error {
+	if len(addSslPair.VirtualHosts) == 0 {
+		return errors.New("NoVirtualHostsProvidedToAddSslPair")
 	}
 
-	sslPairCertificate := addSslPair.Certificate
-	sslCertificates, err := sslQueryRepo.SslCertificatesFactory(
-		sslPairCertificate.Certificate,
-	)
+	vhostQueryRepo := vhostInfra.VirtualHostQueryRepo{}
+	vhosts, err := vhostQueryRepo.Get()
 	if err != nil {
-		return err
+		return errors.New("FailedToGetVhosts")
 	}
 
-	newSsl, err := sslQueryRepo.SslPairFactory(
-		addSslPair.Hostname,
-		addSslPair.Key,
-		sslCertificates,
-	)
-	if err != nil {
-		return err
+	vhostsWithoutAliases := []valueObject.Fqdn{}
+	for _, vhost := range vhosts {
+		if vhost.Type.String() == "alias" {
+			continue
+		}
+
+		if slices.Contains(addSslPair.VirtualHosts, vhost.Hostname) {
+			vhostsWithoutAliases = append(vhostsWithoutAliases, vhost.Hostname)
+		}
 	}
 
-	isChainedCert := true
-	if len(newSsl.ChainCertificates) == 1 {
-		isChainedCert = false
+	firstVhostStr := addSslPair.VirtualHosts[0].String()
+	firstVhostCertFilePath := pkiConfDir + "/" + firstVhostStr + ".crt"
+	firstVhostCertKeyFilePath := pkiConfDir + "/" + firstVhostStr + ".key"
```

Comment:
E se o primeiro elemento da addSslPair.VirtualHosts for um aliases? Você não deveria usar o vhostsWithoutAliases aqui?

Reply:
Tem toda razão!
Ajustado.